src/clojure/clojang/jinterface/otp/messaging.clj

```diff
@@ -108,16 +168,17 @@
   (whereis [this mbox-name]
     "Determine the pid corresponding to a registered name on this node."))
 
-(extend-type OtpMbox MboxObject
+(extend-type OtpMbox
+  MboxObject
   (close [this]
     (.close this))
   (equal? [this other-obj]
     (.equals this other-obj))
   (exit
     ([this reason]
-      (.exit this))
-    ([this recip-pid reason]
-      (.exit recip-pid reason)))
+     (.exit this ^OtpErlangObject reason))
+    ([this ^OtpErlangPid recip-pid ^OtpErlangObject reason]
```

Comment:
I wonder ... can we define our own core.typed type here? E.g., ``OtpStrOrPid``, etc.? And then annotate with that?

(A question for future pondering and/or experimentation ... not a blocker for merge :-))